app/ts/components/pages/SettingsView.tsx

```diff
@@ -231,8 +230,29 @@ const RpcSummary = ({ entries, info }: { entries: RpcEntries, info: RpcEntry })
 				<div>{ info.httpsRpc }</div>
 			</div>
 			<div class = 'actions'>
-				<ConfigureRpcConnection rpcEntries = { entries } rpcInfo = { info } />
+				<ConfigureRpcConnection key = { infoHash } rpcInfo = { info } />
 			</div>
 		</li>
 	)
 }
+
+export function useRpcConnectionsList() {
+	const entries = useSignal<RpcEntries>([])
+
+	const trackRpcListChanges = (message: unknown) => {
+		const maybeParsed = MessageToPopup.safeParse(message)
+		if (!maybeParsed.success) return
```

Comment:
I think we want to throw error here as the message should not be anything else?

Reply:
now throws 'popup message is malformed' if unable to parse

Reply:
you can make this cleaner by just using `.parse` instead of using safeParse, then you don't have to check for success as it throws with a good error right away